code/modules/power/supermatter/supermatter.dm

```diff
@@ -630,6 +642,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			power_transmission_bonus += gas_comp[gasID] * gas_trans[gasID] * (isnull(transit_mod[gasID]) ? 1 : transit_mod[gasID])
 		power_transmission_bonus *= h2obonus
 
+		//Miasma is really just microscopic particulate. It gets consumed like anything else that touches the crystal.
+		if(gas_comp[/datum/gas/miasma])
+			var/miasma_pp = env.return_pressure() * gas_comp[/datum/gas/miasma]
+			var/consumed_miasma = clamp(((miasma_pp - MIASMA_CONSUMPTION_PP) / (miasma_pp + MIASMA_PRESSURE_SCALING)) + gasmix_power_ratio, MIASMA_CONSUMPTION_RATIO_MIN, MIASMA_CONSUMPTION_RATIO_MAX) * gas_comp[/datum/gas/miasma] * combined_gas
```

Comment:
You have brought partial pressures to my household, also please elaborate on the intent of this math.

Reply:
The math is intended to make the amount of miasma consumed by the crystal scale with the partial pressure of the miasma and the gasmix_power_ratio of the crystal. I really just picked the partial pressure bit because I wanted the ratio to approach 1 as the partial pressure approaches infinity. The bit in the numerator is in case someone wants to make miasma not consume below some partial pressure. The gasmix power ratio bit is so other gases can have _some_ effect on miasma consumption. Not gonna lie, I just threw it in. The clamps are to prevent getting consumed mole values outside of [0, 1].

Reply:
why use env.return_pressure and not remove.return_pressure?

Reply:
`env.return_pressure()` has the more accurate pressure value.

Reply:
Yeah but you are using the moles that are inside the remove gas mixture, not the env one, so you are calculating the miasma_pp from a mixture that has a different miasma partial pressure, or am I seeing this wrong?

Reply:
You are seeing it right. The removed gas mixture was just taken from the env mixture. Since the composition by ratio of the gases in both mixtures are identical and the pressure of the env mixture is the more accurate pressure I chose to use the pressure of the env mix and the composition calculated from the removed mix.

Reply:
Oh cool, didn't know this and now I know. Thanks!